From 6ad2198e72b2ad15f866b4e58ad2111325d00df0 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Mon, 1 Jun 2026 21:50:30 +0200
Subject: [PATCH 07/22] Module Discord Fix Stop Server task

---
 .../java/net/essentialsx/discord/EssentialsDiscord.java     | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/EssentialsDiscord/src/main/java/net/essentialsx/discord/EssentialsDiscord.java b/EssentialsDiscord/src/main/java/net/essentialsx/discord/EssentialsDiscord.java
index bda1bb69b..00881ae12 100644
--- a/EssentialsDiscord/src/main/java/net/essentialsx/discord/EssentialsDiscord.java
+++ b/EssentialsDiscord/src/main/java/net/essentialsx/discord/EssentialsDiscord.java
@@ -25,6 +25,7 @@ public class EssentialsDiscord extends JavaPlugin implements IEssentialsModule {
     private DiscordSettings settings;
     private boolean isPAPI = false;
     private boolean isEssentialsChat = false;
+    private boolean isDisabled = false;
 
     @Override
     public void onEnable() {
@@ -82,6 +83,7 @@ public class EssentialsDiscord extends JavaPlugin implements IEssentialsModule {
 
     public void onReload() {
         if (jda != null && !jda.isInvalidStartup()) {
+            isDisabled = false;
             jda.updateListener();
             jda.updatePresence();
             jda.updatePrimaryChannel();
@@ -90,6 +92,10 @@ public class EssentialsDiscord extends JavaPlugin implements IEssentialsModule {
         }
     }
 
+    public boolean isDisabled() {
+        return isDisabled || (jda != null && jda.isInvalidStartup());
+    }
+
     public boolean isInvalidStartup() {
         return jda != null && jda.isInvalidStartup();
     }
-- 
2.54.0.windows.1

