From 1993548631c9ff6ff80d13e6e7aa05a4a515a95d Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Wed, 9 Jul 2025 14:48:57 +0200
Subject: [PATCH 12/12] Downgrade spongepowered yaml

---
 Essentials/build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Essentials/build.gradle b/Essentials/build.gradle
index 4e04f0932..c16685099 100644
--- a/Essentials/build.gradle
+++ b/Essentials/build.gradle
@@ -10,7 +10,7 @@ dependencies {
 
     api 'io.papermc:paperlib:1.0.6'
 
-    implementation 'org.spongepowered:configurate-yaml:4.2.0'
+    implementation 'org.spongepowered:configurate-yaml:4.1.2'
     implementation 'org.checkerframework:checker-qual:3.49.0'
     implementation 'nu.studer:java-ordered-properties:1.0.4'
 
-- 
2.50.0.windows.1

