From 4f2083eb4cfdca76704c539f732e41b02a399619 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Wed, 30 Jul 2025 12:10:49 +0200
Subject: [PATCH 12/12] Downgrade spongepowered yaml

---
 Essentials/build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Essentials/build.gradle b/Essentials/build.gradle
index bf60d7a58..3278272fa 100644
--- a/Essentials/build.gradle
+++ b/Essentials/build.gradle
@@ -10,7 +10,7 @@ dependencies {
 
     api 'io.papermc:paperlib:1.0.6'
 
-    implementation 'org.spongepowered:configurate-yaml:4.2.0'
+    implementation 'org.spongepowered:configurate-yaml:4.1.2'
     implementation 'org.checkerframework:checker-qual:3.49.0'
     implementation 'nu.studer:java-ordered-properties:1.0.4'
 
-- 
2.50.1.windows.1

