From e4b4a8a2c2e211a385fbb659841d7199d9f4a49b Mon Sep 17 00:00:00 2001
From: Bierque Euphyllia <bierque.euphyllia@gmail.com>
Date: Mon, 9 Feb 2026 00:20:20 +0100
Subject: [PATCH 12/14] Downgrade spongepowered yaml

---
 Essentials/build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Essentials/build.gradle b/Essentials/build.gradle
index c33de4218..8f0c6f34b 100644
--- a/Essentials/build.gradle
+++ b/Essentials/build.gradle
@@ -10,7 +10,7 @@ dependencies {
 
     api 'io.papermc:paperlib:1.0.6'
 
-    implementation 'org.spongepowered:configurate-yaml:4.2.0'
+    implementation 'org.spongepowered:configurate-yaml:4.1.2'
     implementation 'org.checkerframework:checker-qual:3.49.0'
     implementation 'nu.studer:java-ordered-properties:1.0.4'
 
-- 
2.52.0.windows.1

