From 8dd920a382426ddd1e361c1d0031eb6444518c85 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Mon, 6 Oct 2025 17:31:45 +0200
Subject: [PATCH 12/14] Downgrade spongepowered yaml

---
 Essentials/build.gradle | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Essentials/build.gradle b/Essentials/build.gradle
index 145edd2a0..11550680f 100644
--- a/Essentials/build.gradle
+++ b/Essentials/build.gradle
@@ -10,7 +10,7 @@ dependencies {
 
     api 'io.papermc:paperlib:1.0.6'
 
-    implementation 'org.spongepowered:configurate-yaml:4.2.0'
+    implementation 'org.spongepowered:configurate-yaml:4.1.2'
     implementation 'org.checkerframework:checker-qual:3.49.0'
     implementation 'nu.studer:java-ordered-properties:1.0.4'
 
-- 
2.50.1.windows.1

