From 97297543816a502f6a1828b22e42b4dae0b01d10 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Sun, 17 May 2026 19:43:54 +0200
Subject: [PATCH 13/21] Fix teleport Folia 26

---
 .../earth2me/essentials/AsyncTeleport.java    | 37 +++++++++++++++++++
 1 file changed, 37 insertions(+)

diff --git a/Essentials/src/main/java/com/earth2me/essentials/AsyncTeleport.java b/Essentials/src/main/java/com/earth2me/essentials/AsyncTeleport.java
index f691d30d8..b5414c409 100644
--- a/Essentials/src/main/java/com/earth2me/essentials/AsyncTeleport.java
+++ b/Essentials/src/main/java/com/earth2me/essentials/AsyncTeleport.java
@@ -4,6 +4,7 @@ import com.earth2me.essentials.api.IAsyncTeleport;
 import com.earth2me.essentials.commands.WarpNotFoundException;
 import com.earth2me.essentials.utils.DateUtil;
 import com.earth2me.essentials.utils.LocationUtil;
+import com.earth2me.essentials.utils.VersionUtil;
 import io.papermc.lib.PaperLib;
 import net.ess3.api.IEssentials;
 import net.ess3.api.IUser;
@@ -186,6 +187,42 @@ public class AsyncTeleport implements IAsyncTeleport {
             targetLoc.setX(LocationUtil.getXInsideWorldBorder(targetLoc.getWorld(), targetLoc.getBlockX()));
             targetLoc.setZ(LocationUtil.getZInsideWorldBorder(targetLoc.getWorld(), targetLoc.getBlockZ()));
         }
+        if (VersionUtil.isFoliaServer()) {
+            ess.getRegionScheduler().run(ess, targetLoc, scheduledTask -> {
+                if (LocationUtil.isBlockUnsafeForUser(ess, teleportee, targetLoc.getWorld(), targetLoc.getBlockX(), targetLoc.getBlockY(), targetLoc.getBlockZ())) {
+                    if (ess.getSettings().isTeleportSafetyEnabled()) {
+                        if (ess.getSettings().isForceDisableTeleportSafety()) {
+                            //The chunk we're teleporting to is 100% going to be loaded here, no need to teleport async.
+                            teleportee.getBase().teleportAsync(targetLoc, cause);
+                        } else {
+                            try {
+                                Location safeDestination = LocationUtil.getSafeDestination(ess, teleportee, targetLoc);
+                                teleportee.getBase().teleportAsync(safeDestination, cause);
+                            } catch (final Exception e) {
+                                future.completeExceptionally(e);
+                                return;
+                            }
+                        }
+                    } else {
+                        future.completeExceptionally(new TranslatableException("unsafeTeleportDestination", targetLoc.getWorld().getName(), targetLoc.getBlockX(), targetLoc.getBlockY(), targetLoc.getBlockZ()));
+                        return;
+                    }
+                } else {
+                    if (ess.getSettings().isForceDisableTeleportSafety()) {
+                        //The chunk we're teleporting to is 100% going to be loaded here, no need to teleport async.
+                        teleportee.getBase().teleportAsync(targetLoc, cause);
+                    } else {
+                        Location roundedDestination = targetLoc;
+                        if (ess.getSettings().isTeleportToCenterLocation()) {
+                            roundedDestination = LocationUtil.getRoundedDestination(targetLoc);
+                        }
+                        teleportee.getBase().teleportAsync(roundedDestination, cause);
+                    }
+                }
+                future.complete(true);
+            });
+            return;
+        }
         PaperLib.getChunkAtAsync(targetLoc.getWorld(), targetLoc.getBlockX() >> 4, targetLoc.getBlockZ() >> 4, true, true).thenAccept(chunk -> {
             Location loc = targetLoc;
             if (LocationUtil.isBlockUnsafeForUser(ess, teleportee, chunk.getWorld(), loc.getBlockX(), loc.getBlockY(), loc.getBlockZ())) {
-- 
2.54.0.windows.1

