From ac9800db7468fae64c1a9671e0226dd995deefe2 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Tue, 26 May 2026 15:49:13 +0200
Subject: [PATCH 14/22] Patch RoleSyncManager

---
 .../discordlink/rolesync/RoleSyncManager.java      | 14 +++++++-------
 1 file changed, 7 insertions(+), 7 deletions(-)

diff --git a/EssentialsDiscordLink/src/main/java/net/essentialsx/discordlink/rolesync/RoleSyncManager.java b/EssentialsDiscordLink/src/main/java/net/essentialsx/discordlink/rolesync/RoleSyncManager.java
index 72353afb7..251e47165 100644
--- a/EssentialsDiscordLink/src/main/java/net/essentialsx/discordlink/rolesync/RoleSyncManager.java
+++ b/EssentialsDiscordLink/src/main/java/net/essentialsx/discordlink/rolesync/RoleSyncManager.java
@@ -30,14 +30,14 @@ public class RoleSyncManager implements Listener {
     private final Map<String, InteractionRole> groupToRoleMap = new HashMap<>();
     private final Map<String, String> roleIdToGroupMap = new HashMap<>();
     private final Semaphore syncSemaphore = new Semaphore(5);
-    private BukkitTask syncTask;
+    private io.papermc.paper.threadedregions.scheduler.ScheduledTask syncTask;
     private int syncCursor = 0;
 
     public RoleSyncManager(final EssentialsDiscordLink ess) {
         this.ess = ess;
         Bukkit.getPluginManager().registerEvents(this, ess);
         onReload();
-        this.syncTask = this.ess.getEss().runTaskTimerAsynchronously(() -> {
+        this.syncTask = this.ess.getEss().getAsyncScheduler().runAtFixedRate(ess, scheduledTask -> {
             if (groupToRoleMap.isEmpty() && roleIdToGroupMap.isEmpty()) {
                 return;
             }
@@ -67,7 +67,7 @@ public class RoleSyncManager implements Listener {
                 final Map.Entry<String, String> entry = entries.get(i);
                 sync(new UUIDPlayer(UUID.fromString(entry.getKey())), entry.getValue(), groupToRoleMapCopy, roleIdToGroupMapCopy, primaryOnly, removeGroups, removeRoles);
             }
-        }, 0, ess.getSettings().getRoleSyncResyncDelay() * 1200L);
+        }, 1, ess.getSettings().getRoleSyncResyncDelay() * 1200L*50, java.util.concurrent.TimeUnit.MILLISECONDS);
     }
 
     public void shutdown() {
@@ -89,7 +89,7 @@ public class RoleSyncManager implements Listener {
                      final boolean primaryOnly, final boolean removeGroups, final boolean removeRoles) {
         final List<String> groups = primaryOnly ?
                 Collections.singletonList(ess.getEss().getPermissionsHandler().getGroup(player)) : ess.getEss().getPermissionsHandler().getGroups(player);
-        ess.getEss().runTaskAsynchronously(() -> {
+        ess.getEss().getAsyncScheduler().runNow(ess, (scheduledTask) -> {
             try {
                 if (!syncSemaphore.tryAcquire(5, TimeUnit.SECONDS)) {
                     return;
@@ -102,7 +102,7 @@ public class RoleSyncManager implements Listener {
                     if (ess.getSettings().isUnlinkOnLeave()) {
                         ess.getLinkManager().removeAccount(ess.getEss().getUser(player), DiscordLinkStatusChangeEvent.Cause.UNSYNC_LEAVE);
                     } else {
-                        ess.getEss().runTaskAsynchronously(() -> unSync(player.getUniqueId(), discordId));
+                        ess.getEss().getAsyncScheduler().runNow(ess, (scheduledTask2) -> unSync(player.getUniqueId(), discordId));
                     }
                     return CompletableFuture.completedFuture(null);
                 }
@@ -165,7 +165,7 @@ public class RoleSyncManager implements Listener {
             return;
         }
 
-        ess.getEss().runTaskAsynchronously(() -> {
+        ess.getEss().getAsyncScheduler().runNow(ess, (scheduledTask) -> {
             try {
                 if (!syncSemaphore.tryAcquire(5, TimeUnit.SECONDS)) {
                     return;
@@ -191,7 +191,7 @@ public class RoleSyncManager implements Listener {
 
     @EventHandler
     public void onJoin(PlayerJoinEvent event) {
-        ess.getEss().runTaskAsynchronously(() -> {
+        ess.getEss().getAsyncScheduler().runNow(ess, (scheduledTask) -> {
             if (ess.getLinkManager().isLinked(event.getPlayer().getUniqueId())) {
                 sync(event.getPlayer().getUniqueId(), ess.getLinkManager().getDiscordId(event.getPlayer().getUniqueId()));
             }
-- 
2.54.0.windows.1

