From 8ed80eeb30ed945048dba9a84329ba6f7c4e8cf0 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Mon, 1 Jun 2026 21:50:33 +0200
Subject: [PATCH 19/22] Fix EssentialsPlayerListener updateCompass

---
 .../earth2me/essentials/EssentialsPlayerListener.java | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java b/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java
index a90a56c47..2a370ddfc 100644
--- a/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java
+++ b/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java
@@ -622,6 +622,17 @@ public class EssentialsPlayerListener implements Listener {
 
         final Location loc = user.getHome(user.getLocation());
         if (loc == null) {
+            if (VersionUtil.isFoliaServer()) {
+                final Player player = user.getBase();
+                final Location bedLocation = player.getRespawnLocation(false);
+                if (bedLocation == null) return;
+                player.teleportAsync(bedLocation, TeleportCause.PLUGIN).thenAccept(success -> {
+                    if (success) {
+                        user.getBase().setCompassTarget(bedLocation);
+                    }
+                });
+                return;
+            }
             PaperLib.getBedSpawnLocationAsync(user.getBase(), false).thenAccept(location -> {
                 if (location != null) {
                     user.getBase().setCompassTarget(location);
-- 
2.54.0.windows.1

