From b57c0055f9eba1052e8ea4f8da1c6cd280f2416b Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Mon, 1 Jun 2026 21:50:33 +0200
Subject: [PATCH 21/22] Fix for crash if a player leaves while someone else

---
 .../essentials/EssentialsPlayerListener.java       | 14 ++++++++------
 1 file changed, 8 insertions(+), 6 deletions(-)

diff --git a/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java b/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java
index 2a370ddfc..1272b2a55 100644
--- a/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java
+++ b/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java
@@ -324,12 +324,14 @@ public class EssentialsPlayerListener implements Listener {
 
         final ArrayList<HumanEntity> viewers = new ArrayList<>(user.getBase().getInventory().getViewers());
         for (final HumanEntity viewer : viewers) {
-            if (viewer instanceof Player) {
-                final User uviewer = ess.getUser((Player) viewer);
-                if (uviewer.isInvSee()) {
-                    uviewer.getBase().closeInventory();
-                }
-            }
+           ess.getEntityScheduler(viewer).run(ess, scheduledTask -> {
+               if (viewer instanceof Player) {
+                   final User uviewer = ess.getUser((Player) viewer);
+                   if (uviewer.isInvSee()) {
+                       uviewer.getBase().closeInventory();
+                   }
+               }
+           }, null);
         }
 
         user.updateActivity(false, AfkStatusChangeEvent.Cause.QUIT);
-- 
2.54.0.windows.1

