From b6c21aaa641ad82aac5b7de2c2aff8ce2f3bb0c5 Mon Sep 17 00:00:00 2001
From: Euphyllia Bierque <bierque.euphyllia@gmail.com>
Date: Tue, 26 May 2026 16:16:59 +0200
Subject: [PATCH 22/22] Try fix issues 27

---
 .../essentials/EssentialsPlayerListener.java        | 13 ++++++++-----
 1 file changed, 8 insertions(+), 5 deletions(-)

diff --git a/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java b/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java
index 1272b2a55..988245c5c 100644
--- a/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java
+++ b/Essentials/src/main/java/com/earth2me/essentials/EssentialsPlayerListener.java
@@ -628,11 +628,14 @@ public class EssentialsPlayerListener implements Listener {
                 final Player player = user.getBase();
                 final Location bedLocation = player.getRespawnLocation(false);
                 if (bedLocation == null) return;
-                player.teleportAsync(bedLocation, TeleportCause.PLUGIN).thenAccept(success -> {
-                    if (success) {
-                        user.getBase().setCompassTarget(bedLocation);
-                    }
-                });
+                ess.getEntityScheduler(player).run(ess, scheduledTask -> {
+                    if (!player.isOnline()) return;
+                    player.teleportAsync(bedLocation, TeleportCause.PLUGIN).thenAccept(success -> {
+                        if (success) {
+                            user.getBase().setCompassTarget(bedLocation);
+                        }
+                    });
+                }, null);
                 return;
             }
             PaperLib.getBedSpawnLocationAsync(user.getBase(), false).thenAccept(location -> {
-- 
2.54.0.windows.1

